server/src/main/java/io/druid/server/http/DatasourcesResource.java

```diff
@@ -496,40 +493,32 @@ public Response getSegmentDataSourceTiers(
     return Response.ok(retVal).build();
   }
 
-  private DruidDataSource getDataSource(final String dataSourceName)
+  private ImmutableDruidDataSource getDataSource(final String dataSourceName)
   {
-    Iterable<DruidDataSource> dataSources =
-        Iterables.concat(
-            Iterables.transform(
-                serverInventoryView.getInventory(),
-                (DruidServer input) -> input.getDataSource(dataSourceName)
-            )
-        );
-
-    List<DruidDataSource> validDataSources = Lists.newArrayList();
-    for (DruidDataSource dataSource : dataSources) {
-      if (dataSource != null) {
-        validDataSources.add(dataSource);
-      }
-    }
-    if (validDataSources.isEmpty()) {
+    List<ImmutableDruidDataSource> dataSources = serverInventoryView
```

Comment:
Why rename "validDataSources" to simply "dataSources"?

Reply:
I don't understand what "valid" means here. In the original code, ```dataSources``` may contain null dataSources, but it means that some servers don't contain the dataSource of the given name, not the dataSource is invalid.